### DISCUSSION TITLE
How do I make a single release of binaries, generated by multiple machines, into Github Releases using Actions?

### DISCUSSION THREAD

**Discussion by WhiteBlackGoose** [community/community#39019] [Actions] [answered]
### Select Topic Area

Question

### Body

So here's a thing. I have CI with three runners for three OS. For each of them, I build an executable. I'd like to make a release in the same CI to include all three executables in it.

I know you can do `gh release create`, but since I have three runners, I need to create it and then edit with two others, right?

How to sync this release among runners? Obviously I can't reliably predict which one finishes first. Or is there a better way? mb another way to host released binaries?

**Comment by airtower-luna** [comment, marked as answer]
The easiest option is probably to sidestep the synchronization by uploading the build results as artifacts and gathering them up in an additional job that creates the release. See: [Passing data between jobs in a workflow](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#passing-data-between-jobs-in-a-workflow)

You might want to set a short retention time for the artifacts when calling `actions/upload-artifact`, simply so they don't clutter the available storage, especially if the executables are large. :wink:
